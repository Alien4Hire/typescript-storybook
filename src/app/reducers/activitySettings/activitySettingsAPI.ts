import { UUID } from '../../../../../app/types';
import { ActivitySettingT } from '../../../../../app/types/activitySettings';
import { apiBase } from '../../api';

type ActivitySettingQueryT = {
  templateId: UUID | undefined;
  templateVersionId: UUID | undefined;
};

type AssignActivityResponseT = {
  id: string;
  template_node_content_id: string;
  name: string;
  description: string;
  activity_settings_id: string;
  createdAt: string;
  updatedAt: string;
}[];

type AssignActivityT = {
  activitySettingId: string;
  templateNodeIds: string[];
};

export const ActivitySettingsService = apiBase.injectEndpoints({
  endpoints: (builder) => ({
    getActivitySettings: builder.query<
      ActivitySettingT[],
      ActivitySettingQueryT
    >({
      query: ({ templateId, templateVersionId }) => ({
        url: `/activitysettings/${templateId}/version/${templateVersionId}`,
      }),
      providesTags: ['ActivitySettings'],
    }),
    addActivitySetting: builder.mutation<ActivitySettingT, ActivitySettingT>({
      query: ({ templateId, templateVersionId, ...activitySetting }) => ({
        url: `/activitysettings/${templateId}/version/${templateVersionId}`,
        method: 'POST',
        body: activitySetting,
        invalidatesTags: ['ActivitySettings'],
      }),
    }),
    deleteActivitySetting: builder.mutation<ActivitySettingT, string>({
      query: (activitySettingId) => ({
        url: `/activitysettings/${activitySettingId}`,
        method: 'DELETE',
        invalidatesTags: ['ActivitySettings'],
      }),
    }),
    assignActivities: builder.mutation<
      AssignActivityResponseT,
      AssignActivityT
    >({
      query: ({ activitySettingId, templateNodeIds }) => ({
        url: `/activitysettings/${activitySettingId}`,
        method: 'PUT',
        body: { templateNodeIds },
      }),
      invalidatesTags: ['ActivitySettings'],
    }),
  }),
});

export const {
  useGetActivitySettingsQuery,
  useAddActivitySettingMutation,
  useDeleteActivitySettingMutation,
  useAssignActivitiesMutation,
} = ActivitySettingsService;
