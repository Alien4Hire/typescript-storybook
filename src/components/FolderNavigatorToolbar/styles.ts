import { Box, Button } from '@mui/material';

import { styled } from '@mui/material/styles';

export const Divider = styled('div')(({ theme }) => ({
  borderRight: `1px solid ${theme.palette.gray.medium}`,
  height: '34px',
  margin: 'auto',
}));

export const Label = styled('div')(({ theme }) => ({
  fontWeight: '600',
  fontSize: '15px',
  letterSpacing: '0.4px',
  color: theme.palette.gray.veryDark,
  fontFamily: "'Source Sans Pro', sans-serif",
}));

export const Wrapper = styled('div')(({ theme }) => ({
  position: 'relative',
  background: theme.palette.gray.veryLight,
  paddingLeft: '56px',
}));

export const FolderForm = styled('div')`
  display: flex;
  align-items: center;
`;

export const FolderAdd = styled(Box, {
  shouldForwardProp: (prop: string) => prop !== 'active',
})(({ active, theme }) => ({
  position: 'absolute',
  top: 0,
  bottom: 0,
  left: 0,
  right: 0,
  width: 'auto',
  padding: '2px',
  backgroundColor: theme.palette?.gray.veryLight,
  display: 'none',
  ...(active && {
    display: 'flex',
    alignItems: 'center',
  }),
}));

export const CsvModalBackdrop = styled(Box)({
  display: 'flex',
  flexDirection: 'column',
  justifyContent: 'center',
  alignItems: 'center',
  backgroundColor: '#f1f1f1',
  borderRadius: '16px',
  padding: '32px 32px 70px',
});

export const UploadListItem = styled(Box)(({ theme }) => ({
  display: 'flex',
  alignItems: 'center',
  padding: '16px 0px',
  borderBottom: `1px solid ${theme.palette.gray.light}`,
}));

export const UploadButton = styled(Button)({
  textTransform: 'none',
  width: 'fit-content',
  marginTop: '16px',
});

export const DownloadTemplateWrapper = styled(Box)({
  display: 'flex',
  alignItems: 'center',
  marginTop: '24px',
  cursor: 'pointer',
});
