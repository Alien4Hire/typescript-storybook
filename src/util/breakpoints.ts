export const mobileScreenMin = 660;
export const mobileScreenMax = 768;
export const tabletScreenMax = 992;
export const computerSmallMonitorMax = 1200;
