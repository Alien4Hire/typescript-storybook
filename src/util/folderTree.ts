import { TemplateNodeOrderT } from '../../../app/types/templateNodes';
import { RecommendedUseEnum } from '../../../app/types/activitySettings';

export type FolderTypeT = 'root' | 'folder' | 'activity' | 'assessment' | 'url';

export type FolderT = {
  id: string;
  parentId?: string;
  name: string;
  sequence?: number;
  type: FolderTypeT;
  children?: FolderT[];
  activitySettingName?: string | null;
  activitySettingRecommendedUse?: keyof typeof RecommendedUseEnum | null;
  activitySettingVisibility?: boolean | null;
};

export type FolderMoveDirectionT = 'up' | 'double-up' | 'down' | 'double-down';

export const addParentIdToFolders = (nodes: FolderT) => {
  return {
    ...nodes,
    children: getChildren(nodes.children, nodes.id),
  };

  function getChildren(
    children?: FolderT[],
    id?: string
  ): FolderT[] | undefined {
    if (children) {
      return children.map((child) => ({
        ...child,
        parentId: id!,
        children: getChildren(child.children, child.id),
      }));
    }
    return;
  }
};

export function getNodeById(nodes: FolderT, id: string) {
  if (nodes.id === id) {
    return nodes;
  }
  let result = null;
  nodes.children?.forEach?.((node) => {
    if (getNodeById(node, id)) {
      result = getNodeById(node, id);
    }
  });
  return result;
}

export function foldersForDropdown({ id, name, children }: FolderT) {
  const step = 0;
  const options = [];

  options.push({
    label: name,
    value: id,
    step,
  });
  getChildren(step + 1, children);
  return options;

  function getChildren(step: number, children?: FolderT[]): void {
    if (children) {
      children.forEach(({ id, name, type, children }) => {
        if (type === 'folder') {
          options.push({
            label: name,
            value: id,
            step,
          });
        }

        if (children) {
          getChildren(step + 1, children);
        }
      });
    }
  }
}

const orderByDirection = {
  up: (position: number) => (position ? position - 1 : 0),
  'double-up': () => 0,
  down: (position: number, length: number) =>
    position + 1 < length ? position + 1 : length - 1,
  'double-down': (_: unknown, length: number) => length,
};

class FolderTree {
  moveFolderItem(
    nodes: FolderT,
    selected: string[],
    direction: FolderMoveDirectionT,
    updatedNodes: TemplateNodeOrderT[]
  ): FolderT {
    const childrens = nodes.children;

    if (!childrens) return nodes;

    const hasSelected = childrens.some(({ id }) => selected.includes(id));

    if (hasSelected) {
      // TODO: rewrite with reduce & without .splice()
      const result = childrens.filter((node) => !selected.includes(node.id));
      const needReverse = ['double-up'].includes(direction);
      const nodesForInsert = needReverse ? childrens.reverse() : childrens;
      let countOfSelected = 0;

      nodesForInsert?.forEach?.((node, index) => {
        if (selected.includes(node.id)) {
          const isTopItem = countOfSelected++ === index;
          let startIndex = orderByDirection[direction](index, childrens.length);
          if (isTopItem && direction === 'up') {
            startIndex = index;
          }
          result.splice(startIndex, 0, node);
        }
      });

      // update all sequence values of the children
      result.forEach((node, index) => {
        updatedNodes.push({
          id: node.id,
          sequence: index + 1,
        });
      });
      return { ...nodes, children: result };
    }

    return {
      ...nodes,
      children: nodes.children?.map((node) =>
        this.moveFolderItem(node, selected, direction, updatedNodes)
      ),
    };
  }

  selectFolderItem(
    folders: FolderT,
    selected: string[],
    checked: boolean,
    nodes: FolderT
  ) {
    const topLevel = folders.children!.map(({ id }) => id);
    const isTop = topLevel.find((item) => item === nodes.id);
    const allNode = this.getChildById(folders, nodes.id);

    const array = checked
      ? isTop
        ? [...allNode]
        : [...selected, ...allNode]
      : selected.filter((value) => !allNode.includes(value));

    return array.filter((v, i) => array.indexOf(v) === i);
  }

  addFolderItem(folders: FolderT, file: Omit<FolderT, 'children'>): FolderT {
    if (folders.id === file.parentId) {
      return {
        ...folders,
        children: [...(folders.children ?? []), file],
      };
    }

    return {
      ...folders,
      children: folders.children?.map((item) => this.addFolderItem(item, file)),
    };
  }

  addFolderItems(
    initialFolders: FolderT,
    files: Omit<FolderT, 'children'>[]
  ): FolderT {
    return files.reduce((folders: FolderT, file) => {
      if (folders.id === file.parentId) {
        return {
          ...folders,
          children: [...(folders.children ?? []), file],
        };
      }

      return {
        ...folders,
        children: folders.children?.map((item) =>
          this.addFolderItem(item, file)
        ),
      };
    }, initialFolders);
  }

  removeFolderItem(folders: FolderT, file: Omit<FolderT, 'children'>): FolderT {
    if (folders.id === file.parentId) {
      return {
        ...folders,
        children: folders.children?.filter((item) => item.id !== file.id),
      };
    }

    return {
      ...folders,
      children: folders.children?.map((item) =>
        this.removeFolderItem(item, file)
      ),
    };
  }

  expandFolderItem(
    folders: FolderT,
    selected: string[],
    expanded: string[],
    nodeId: string[]
  ) {
    const topLevel = folders.children!.map(({ id }) => id);
    const topLevelItems = topLevel.filter((item) => nodeId.includes(item));
    const severalTopFoldersSelected =
      topLevel.filter(
        (item) => nodeId.includes(item) || selected.includes(item)
      ).length > 1;
    const isSeveralTopFolders =
      topLevelItems.length > 1 || severalTopFoldersSelected;

    if (isSeveralTopFolders) {
      const openedTopLevel = topLevelItems.filter(
        (item) => !expanded.includes(item)
      );
      return { selected: [], expanded: openedTopLevel };
    }

    const isCloseFolder = !topLevelItems.length;
    if (isCloseFolder) {
      return { selected: [], expanded: [] };
    } else {
      const isOpenNewFolderFromOther = topLevelItems.length === nodeId.length;
      let selected;
      if (isOpenNewFolderFromOther && severalTopFoldersSelected) {
        selected = [];
      }
      return { selected, expanded: nodeId };
    }
  }

  computeCanReorganize(folders: FolderT, selected: string[]) {
    if (!selected.length) return false;

    const selectedLevels = new Set();
    let hasUnFilledSubFolders = false;

    const runAtNodes = (nodes: FolderT) => {
      nodes?.children?.forEach((node) => {
        const isFolder = node.type === 'folder';
        const isSelected = selected.includes(node.id);
        isSelected && selectedLevels.add(node.parentId);

        if (!isFolder) return;

        if (isSelected) {
          // if selected folder
          const allNodes = this.getChildById(nodes, node.id);
          const isFullSelected = allNodes.every((node) =>
            selected.includes(node)
          );
          if (!isFullSelected) {
            hasUnFilledSubFolders = true;
          }
        } else {
          // if unselected folder
          runAtNodes(node);
        }
      });
    };
    runAtNodes(folders);
    const isValid = !hasUnFilledSubFolders && selectedLevels.size === 1;
    return isValid;
  }

  getChildById(node: FolderT, id: string) {
    let array: string[] = [];

    function getAllChild(nodes: FolderT | null) {
      if (!nodes) return [];
      array.push(nodes.id);
      nodes.children?.forEach?.((node) => {
        array = [...array, ...getAllChild(node)];
        array = array.filter((v, i) => array.indexOf(v) === i);
      });
      return array;
    }

    return getAllChild(getNodeById(node, id));
  }
}

const folderTree = new FolderTree();

export default folderTree;
