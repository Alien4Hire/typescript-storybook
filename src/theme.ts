import { createTheme } from '@mui/material/styles';
import mlColors from './util/ml-cdl-colors';
import {
  computerSmallMonitorMax,
  mobileScreenMax,
  mobileScreenMin,
  tabletScreenMax,
} from './util/breakpoints';

declare module '@mui/material/Typography' {
  interface TypographyPropsVariantOverrides {
    title: true;
    textLarge: true;
    textBasic: true;
    textSmall: true;
    textSub: true;
    buttonMedium: true;
    buttonSmall: true;
    buttonLarge: true;
  }
}

//UX variants definitions can be found at https://www.figma.com/file/bEonWYX4lesmqlFY6MqWCt/Course-Assembly-MVP?node-id=5540%3A249648
//Custom created variant names for typographies

declare module '@mui/material/styles/createTypography' {
  interface Typography {
    formTitle?: React.CSSProperties;
    formSubtitle?: React.CSSProperties;
    appTitle?: React.CSSProperties;
    title?: React.CSSProperties;
    textLarge?: React.CSSProperties;
    textBasic?: React.CSSProperties;
    textSmall?: React.CSSProperties;
    textSub?: React.CSSProperties;
    buttonMedium?: React.CSSProperties;
    buttonSmall?: React.CSSProperties;
    buttonLarge?: React.CSSProperties;
  }

  // allow configuration using `createMuiTheme`
  interface TypographyOptions {
    title?: React.CSSProperties;
    textLarge?: React.CSSProperties;
    textBasic?: React.CSSProperties;
    textSmall?: React.CSSProperties;
    textSub?: React.CSSProperties;
    buttonMedium?: React.CSSProperties;
    buttonSmall?: React.CSSProperties;
    buttonLarge?: React.CSSProperties;
  }
}

declare module '@mui/material/styles' {
  interface Theme {}
  interface ThemeOptions {}
  interface Palette {
    specials: Palette['primary'];
    aqua: Palette['primary'];
    black: Palette['primary'];
    blue: Palette['primary'];
    gray: Palette['primary'];
    green: Palette['primary'];
    magenta: Palette['primary'];
    orange: Palette['primary'];
    purple: Palette['primary'];
    red: Palette['primary'];
    teal: Palette['primary'];
    white: Palette['primary'];
    yellow: Palette['primary'];
  }
  interface PaletteOptions {
    specials: PaletteOptions['primary'];
    aqua: PaletteOptions['primary'];
    black: PaletteOptions['primary'];
    blue: PaletteOptions['primary'];
    gray: PaletteOptions['primary'];
    green: PaletteOptions['primary'];
    magenta: PaletteOptions['primary'];
    orange: PaletteOptions['primary'];
    purple: PaletteOptions['primary'];
    red: PaletteOptions['primary'];
    teal: PaletteOptions['primary'];
    white: PaletteOptions['primary'];
    yellow: PaletteOptions['primary'];
  }

  interface PaletteColor {
    macmillanRed?: string;
    offWhite?: string;
    veryLight?: string;
    medium?: string;
    veryDark?: string;
    fullPageDialogShadow?: string;
  }

  interface SimplePaletteColorOptions {
    macmillanRed?: string;
    offWhite?: string;
    veryLight?: string;
    medium?: string;
    veryDark?: string;
    fullPageDialogShadow?: string;
  }
}

export const theme = createTheme({
  palette: {
    background: {
      default: mlColors.off_white,
      paper: mlColors.pure_white,
    },
    primary: {
      main: mlColors.aqua,
    },
    secondary: {
      main: mlColors.aqua_dark,
    },
    specials: {
      main: mlColors.macmillan_red,
      macmillanRed: mlColors.macmillan_red,
      fullPageDialogShadow: 'rgba(0, 0, 0, 0.5)',
    },
    aqua: {
      main: mlColors.aqua,
      veryLight: mlColors.aqua_very_light,
      light: mlColors.aqua_light,
      medium: mlColors.aqua_medium,
      dark: mlColors.aqua_dark,
      veryDark: mlColors.dark_navy,
    },
    black: {
      main: mlColors.black,
    },
    blue: {
      main: mlColors.blue,
      veryLight: mlColors.blue_very_light,
      light: mlColors.blue_light,
      medium: mlColors.blue_medium,
      dark: mlColors.blue_dark,
    },
    gray: {
      main: mlColors.gray,
      veryLight: mlColors.gray_very_light,
      light: mlColors.gray_light,
      medium: mlColors.gray_medium,
      dark: mlColors.gray_dark,
      veryDark: mlColors.gray_very_dark,
    },
    green: {
      main: mlColors.green,
      veryLight: mlColors.green_very_light,
      light: mlColors.green_light,
      medium: mlColors.green_medium,
      dark: mlColors.green_dark,
    },
    magenta: {
      main: mlColors.magenta,
      veryLight: mlColors.magenta_very_light,
      light: mlColors.magenta_light,
      medium: mlColors.magenta_medium,
      dark: mlColors.magenta_dark,
    },
    orange: {
      main: mlColors.orange,
      veryLight: mlColors.orange_very_light,
      light: mlColors.orange_light,
      medium: mlColors.orange_medium,
      dark: mlColors.orange_dark,
    },
    purple: {
      main: mlColors.purple,
      veryLight: mlColors.purple_very_light,
      light: mlColors.purple_light,
      medium: mlColors.purple_medium,
      dark: mlColors.purple_dark,
    },
    red: {
      main: mlColors.red,
      veryLight: mlColors.red_very_light,
      light: mlColors.red_light,
      medium: mlColors.red_medium,
      dark: mlColors.red_dark,
    },
    teal: {
      main: mlColors.teal,
      veryLight: mlColors.teal_very_light,
      light: mlColors.teal_light,
      medium: mlColors.teal_medium,
      dark: mlColors.teal_dark,
      veryDark: mlColors.teal_very_dark,
    },
    white: {
      main: mlColors.pure_white,
      offWhite: mlColors.off_white,
    },
    yellow: {
      main: mlColors.yellow,
      veryLight: mlColors.yellow_very_light,
      light: mlColors.yellow_light,
      medium: mlColors.yellow_medium,
      dark: mlColors.yellow_dark,
    },
  },
  breakpoints: {
    values: {
      xs: 0,
      sm: mobileScreenMin,
      md: mobileScreenMax,
      lg: tabletScreenMax,
      xl: computerSmallMonitorMax,
    },
  },
  spacing: 8,
  typography: {
    fontSize: 14,
    fontFamily: [
      '-apple-system',
      'BlinkMacSystemFont',
      '"Segoe UI"',
      'Roboto',
      'Oxygen',
      'Ubuntu',
      'Cantarell',
      '"Fira Sans"',
      '"Droid Sans"',
      '"Helvetica Neue"',
      'Source Sans Pro',
      'sans-serif',
    ].join(','),
    allVariants: {
      fontWeight: 'normal',
    },
    button: {
      fontSize: 14,
      textTransform: 'uppercase',
    },
    h1: {
      color: mlColors.black,
      fontSize: 32,
      fontFamily: 'Source Sans Pro',
      fontWeight: 600,
    },
    h2: {
      color: mlColors.black,
      fontSize: 24,
      fontFamily: 'Source Sans Pro',
      fontWeight: 600,
    },
    h3: {
      color: mlColors.black,
      fontSize: 18,
      fontFamily: 'Source Sans Pro',
      fontWeight: 600,
    },
    h4: {
      color: mlColors.black,
      fontSize: 16,
      fontFamily: 'Source Sans Pro',
      fontWeight: 600,
    },
    title: {
      color: mlColors.black,
      fontSize: 28,
      fontFamily: 'Source Sans Pro',
      fontWeight: 200,
      fontStyle: 'normal',
    },
    textLarge: {
      color: mlColors.black,
      fontSize: 18,
      fontFamily: 'Source Sans Pro',
      fontWeight: 400,
      fontStyle: 'normal',
    },
    textBasic: {
      color: mlColors.black,
      fontSize: 16,
      fontFamily: 'Source Sans Pro',
      fontWeight: 400,
      fontStyle: 'normal',
    },
    textSmall: {
      color: mlColors.black,
      fontSize: 14,
      fontFamily: 'Source Sans Pro',
      fontWeight: 400,
      fontStyle: 'normal',
    },
    textSub: {
      color: mlColors.black,
      fontSize: 13,
      fontFamily: 'Source Sans Pro',
      fontWeight: 400,
      fontStyle: 'normal',
    },
    buttonLarge: {
      color: mlColors.black,
      fontSize: 15,
      fontFamily: 'Source Sans Pro',
      fontWeight: 400,
      fontStyle: 'normal',
    },
    buttonMedium: {
      color: mlColors.black,
      fontSize: 18,
      fontFamily: 'Source Sans Pro',
      fontWeight: 400,
      fontStyle: 'normal',
    },
    buttonSmall: {
      color: mlColors.black,
      fontSize: 12,
      fontFamily: 'Source Sans Pro',
      fontWeight: 400,
      fontStyle: 'normal',
    },
  },
  components: {
    MuiAlert: {
      styleOverrides: {
        filledError: {
          background: mlColors.red_very_light,
          color: mlColors.black,
          border: `1px solid ${mlColors.red}`,
        },
        filledInfo: {
          background: mlColors.blue_very_light,
          color: mlColors.black,
          border: `1px solid ${mlColors.blue}`,
        },
        filledSuccess: {
          background: mlColors.green_very_light,
          color: mlColors.black,
          border: `1px solid ${mlColors.green}`,
        },
        filledWarning: {
          background: mlColors.yellow_very_light,
          color: mlColors.black,
          border: `1px solid ${mlColors.yellow}`,
        },
      },
    },
    // MuiButton: {
    //   styleOverrides: {
    //     root: {
    //       fontSize: '1rem',
    //       textTransform: 'none',
    //       height: '35px',
    //       padding: '0 15px',
    //       margin: '5px',
    //       '&:focus': {
    //         boxShadow: '0 0 6px rgba(0, 117, 142, 1)',
    //         border: '1px solid #FFFFFF',
    //       },
    //       '&:disabled': {
    //         background: '#c8c8c8',
    //         color: '#f3f3f3',
    //       },
    //     },
    //     textSecondary: {
    //       fontSize: '0.9375rem',
    //       height: '34px',
    //       lineHeight: '33px',
    //       padding: '0 15px',
    //       borderRadius: '3px',
    //       '-webkit-border-radius': '3px',
    //       '-ms-border-radius': '3px',
    //       '-moz-border-radius': '3px',
    //       textDecoration: 'none',
    //       color: '#00758e',
    //       cursor: 'pointer',
    //       background: 'transparent',
    //       border: '1px solid #00758e',
    //       overflow: 'hidden',
    //     },
    //     containedPrimary: {
    //       '&:focus': {
    //         boxShadow: '0 0 6px rgba(0, 117, 142, 1)',
    //         border: '1px solid #FFFFFF',
    //       },
    //       '&:disabled': {
    //         background: '#c8c8c8',
    //         color: '#f3f3f3',
    //       },
    //     },
    //     outlinedSecondary: {
    //       '&:focus': {
    //         boxShadow: '0 0 6px rgba(0, 117, 142, 1)',
    //         border: '1px solid #00758e',
    //       },
    //       '&:disabled': {
    //         color: '#989898',
    //         fill: '#c8c8c8',
    //         backgroundColor: '#f3f3f3',
    //       },
    //     },
    //   },
    // },
    MuiButtonBase: {
      defaultProps: {
        disableRipple: true, // No more ripple, on the whole application!
        focusRipple: false,
      },
    },
  },
});
