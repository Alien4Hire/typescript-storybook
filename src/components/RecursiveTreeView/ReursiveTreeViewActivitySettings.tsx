import React from 'react';

import { Box, Typography, Tooltip } from '@mui/material';

import { truncate } from '../../util/string';
import { FolderT } from '../../util/folderTree';
import { theme } from '../../theme';
import { ActivitySettingInfoWrapper } from './RecursiveTreeView.styles';

type MlRecursiveTreeViewActivitySettingsPropsT = {
  nodes: FolderT;
  testId?: string;
};

const RecursiveTreeViewActivitySettings = ({
  nodes,
  testId = '',
}: MlRecursiveTreeViewActivitySettingsPropsT) => {
  return (
    <Box
      data-testid={testId}
      sx={{
        display: 'flex',
        alignItems: 'center',
        justifyContent: 'space-between',
      }}
    >
      <ActivitySettingInfoWrapper>
        <Typography
          variant="textBasic"
          sx={{
            color: theme.palette.gray.veryDark,
            lineHeight: '24px',
          }}
        >
          Activity Setting:
        </Typography>
        <Tooltip title={`${nodes.activitySettingName}`}>
          <Typography variant="h4">
            {truncate(nodes.activitySettingName || '', 25)}
          </Typography>
        </Tooltip>
      </ActivitySettingInfoWrapper>
      <ActivitySettingInfoWrapper>
        <Typography
          variant="textBasic"
          sx={{
            color: theme.palette.gray.veryDark,
            lineHeight: '24px',
          }}
        >
          Recommend Use:
        </Typography>
        <Typography variant="h4">
          {nodes.activitySettingRecommendedUse}
        </Typography>
      </ActivitySettingInfoWrapper>
      <ActivitySettingInfoWrapper>
        <Typography
          variant="textBasic"
          sx={{
            color: theme.palette.gray.veryDark,
            lineHeight: '24px',
          }}
        >
          Visibility:
        </Typography>
        <Typography variant="h4">
          {nodes.activitySettingVisibility ? 'Available' : 'Hidden'}
        </Typography>
      </ActivitySettingInfoWrapper>
    </Box>
  );
};

export default RecursiveTreeViewActivitySettings;
