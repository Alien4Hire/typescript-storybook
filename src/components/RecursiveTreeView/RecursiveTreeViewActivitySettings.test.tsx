import { render, screen } from '@testing-library/react';
import { ThemeProvider, StyledEngineProvider } from '@mui/material/styles';

import ReursiveTreeViewActivitySettings from './ReursiveTreeViewActivitySettings';

import { theme } from '../../theme';
import { FolderTypeT } from '../../util/folderTree';
import { RecommendedUseEnum } from '../../../../app/types/activitySettings';

import '../../setupTests';

const baseProps = {
  nodes: {
    id: 'aoibsdioasbd',
    name: 'Intro quiz',
    type: 'activity' as FolderTypeT,
    activitySettingName: 'Activity Setting 1',
    activitySettingRecommendedUse: 'none' as keyof typeof RecommendedUseEnum,
    activitySettingVisibility: true,
  },
};

const renderComponent = (customProps?: { testId?: string }) => {
  const props = { ...baseProps, ...customProps };

  return render(
    <StyledEngineProvider injectFirst>
      <ThemeProvider theme={theme}>
        <ReursiveTreeViewActivitySettings {...props} />
      </ThemeProvider>
    </StyledEngineProvider>
  );
};

describe('<ReursiveTreeViewActivitySettings />', () => {
  it('renders without crashing', () => {
    const view = renderComponent();
    expect(view).toMatchSnapshot();
  });

  it('should have the Activity Setting name', () => {
    renderComponent();
    expect(
      screen.getByText(baseProps.nodes.activitySettingName)
    ).toBeInTheDocument();
  });

  it('should have the Activity Setting recommended use', () => {
    renderComponent();
    expect(
      screen.getByText(baseProps.nodes.activitySettingRecommendedUse)
    ).toBeInTheDocument();
  });

  it('should have the Activity Setting visibility', () => {
    renderComponent();
    expect(
      screen.getByText(
        baseProps.nodes.activitySettingVisibility ? 'Available' : 'Hidden'
      )
    ).toBeInTheDocument();
  });

  it('should have a data-test-id', () => {
    const testId = 'test-recursive-tree-view-activity-setting';
    renderComponent({ testId });
    const element = screen.getByTestId(testId);
    expect(element).toBeInTheDocument();
  });
});
