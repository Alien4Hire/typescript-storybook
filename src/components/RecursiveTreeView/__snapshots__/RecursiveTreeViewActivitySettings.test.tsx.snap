// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ReursiveTreeViewActivitySettings /> renders without crashing 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c2 {
  margin: 0;
  color: #080808;
  font-size: 16px;
  font-family: Source Sans Pro;
  font-weight: 400;
  font-style: normal;
  color: #383838;
  line-height: 24px;
}

.c3 {
  margin: 0;
  color: #080808;
  font-size: 16px;
  font-family: Source Sans Pro;
  font-weight: 600;
  line-height: 1.235;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin-right: 10px;
}

<body>
    <div>
      <div
        class="c0 MuiBox-root"
        data-testid=""
      >
        <div
          class="c1 MuiBox-root"
        >
          <span
            class="c2 MuiTypography-root MuiTypography-textBasic"
          >
            Activity Setting:
          </span>
          <h4
            aria-label="Activity Setting 1"
            class="c3 MuiTypography-root MuiTypography-h4"
            data-mui-internal-clone-element="true"
          >
            Activity Setting 1
          </h4>
        </div>
        <div
          class="c1 MuiBox-root"
        >
          <span
            class="c2 MuiTypography-root MuiTypography-textBasic"
          >
            Recommend Use:
          </span>
          <h4
            class="c3 MuiTypography-root MuiTypography-h4"
          >
            none
          </h4>
        </div>
        <div
          class="c1 MuiBox-root"
        >
          <span
            class="c2 MuiTypography-root MuiTypography-textBasic"
          >
            Visibility:
          </span>
          <h4
            class="c3 MuiTypography-root MuiTypography-h4"
          >
            Available
          </h4>
        </div>
      </div>
    </div>
  </body>,
  "container": .c2 {
  margin: 0;
  color: #080808;
  font-size: 16px;
  font-family: Source Sans Pro;
  font-weight: 400;
  font-style: normal;
  color: #383838;
  line-height: 24px;
}

.c3 {
  margin: 0;
  color: #080808;
  font-size: 16px;
  font-family: Source Sans Pro;
  font-weight: 600;
  line-height: 1.235;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin-right: 10px;
}

<div>
    <div
      class="c0 MuiBox-root"
      data-testid=""
    >
      <div
        class="c1 MuiBox-root"
      >
        <span
          class="c2 MuiTypography-root MuiTypography-textBasic"
        >
          Activity Setting:
        </span>
        <h4
          aria-label="Activity Setting 1"
          class="c3 MuiTypography-root MuiTypography-h4"
          data-mui-internal-clone-element="true"
        >
          Activity Setting 1
        </h4>
      </div>
      <div
        class="c1 MuiBox-root"
      >
        <span
          class="c2 MuiTypography-root MuiTypography-textBasic"
        >
          Recommend Use:
        </span>
        <h4
          class="c3 MuiTypography-root MuiTypography-h4"
        >
          none
        </h4>
      </div>
      <div
        class="c1 MuiBox-root"
      >
        <span
          class="c2 MuiTypography-root MuiTypography-textBasic"
        >
          Visibility:
        </span>
        <h4
          class="c3 MuiTypography-root MuiTypography-h4"
        >
          Available
        </h4>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
